lib/daru/io/importers/json.rb

```diff
@@ -0,0 +1,174 @@
+        #
+        #   df
+        #
+        #   #=> #<Daru::DataFrame(61x4)>
+        #   #=>               name     season     number    RunTime
+        #   #=>      10 Winter is           1          1         60
+        #   #=>      11 The Kingsr          1          2         60
+        #   #=>      12  Lord Snow          1          3         60
+        #   #=>      13 Cripples,           1          4         60
+        #   #=>      14 The Wolf a          1          5         60
+        #   #=>      15 A Golden C          1          6         60
+        #   #=>      16 You Win or          1          7         60
+        #   #=>      17 The Pointy          1          8         60
+        #   #=>      18     Baelor          1          9         60
+        #   #=>      19 Fire and B          1         10         60
+        #   #=>      20 The North           2          1         60
+        #   #=>      21 The Night           2          2         60
+        #   #=>      22 What is De          2          3         60
+        #   #=>      23 Garden of           2          4         60
+        #   #=>      24 The Ghost           2          5         60
+        #   #=>     ...        ...        ...        ...        ...
+        def initialize(input, *arrays, order: nil, index: nil, **hashes)
+          super(binding)
+          @data       = []
+          @auto_order = []
+        end
+
+        def call
+          json = read_json
+
+          if @hashes.empty?
+            if @arrays.size == 1
+              @data = get_xpath(json, @arrays.first)
+            elsif @arrays.empty?
+              parse_without_hash json
+            else
+              parse_with_array json
+            end
+          else
+            parse_with_array json unless @arrays.empty?
+            parse_with_hash json
+          end
+
+          @order ||= @auto_order
+          Daru::DataFrame.new(@data, order: @order, index: @index)
+        end
+
+        private
+
+        def read_json
+          # Checks if input is a remote JSON file, local JSON file,
+          # API JSON response or JSON string
+          if @input.is_a? String
+            if @input.start_with?('http') || @input.end_with?('.json')
+              # A local or remote JSOn file
+              ::JSON.parse(open(@input).read)
+            else
+              # A JSON string
+              ::JSON.parse(@input)
+            end
+          else
+            # A JSON response
+            @input
+          end
+        end
+
+        def get_xpath(json, xpath)
+          case xpath
+          when String
+            JsonPath.on(json, xpath)
+          when Array, Hash
+            xpath
+          when nil
+            nil
+          end
+        end
+
+        def parse_with_array(json)
+          @data        += @arrays.map { |xpath| get_xpath(json, xpath) }
+          @index        = get_xpath(json, @index)
+          @auto_order  += @arrays.map { |xpath| xpath.split('..').last }
```

Comment:
Let's be as simple as possible. As DF supports unnamed columns, we can just go with it.